server.js

```diff
@@ -253,34 +267,115 @@ async function generateNicknameTag(db, nickname) {
       if (!tokenValidation.success) {
          return res.json(tokenValidation);
       }
-      const userGettingRequest = await getUserUniqidByNicknameAndTag(db, req.body.userReqestedToAddNickname, req.body.userReqestedToAddTag)
-      if (!userGettingRequest) {
+      const userGettingRequestUniqid = await getUserUniqidByNicknameAndTag(db, req.body.userReqestedToAddNickname, req.body.userReqestedToAddTag)
+      if (!userGettingRequestUniqid) {
          return res.json({
             success: false,
             msg: 'second user not found in database'
          });
       }
-      if (userGettingRequest === userRequesting) {
+      if (userRequesting.uniqid === userGettingRequestUniqid) {
          return res.json({
             success: false,
-            msg: 'you can not yourself to friends'
+            msg: 'you can not add yourself to friends'
          });
       }
-      const addedUserResponse = await addUserToFriendRequest(db, userRequesting, userGettingRequest)
+      const addedUserResponse = await addUserToFriendRequest(db, userRequesting.uniqid, userGettingRequestUniqid)
       return res.send(addedUserResponse);
    });
 
+   app.post('/acceptFriend', async (req, res) => {
+      if (!req.body.token || !req.body.tokenSelector || !req.body.friendNickname || !req.body.friendsTag) {
+         return res.json({
+            success: false,
+            msg: 'incomplete query'
+         });
+      }
+      const user = await getUserByTokenSelector(db, req.body.tokenSelector)
+      if (!user) {
+         return res.json({
+            success: false,
+            msg: 'User trying to add friend was not found in database'
+         });
+      }
+      const tokenValidation = await validateUserToken(db, user.uniqid, req.body.token)
+      if (!tokenValidation.success) {
+         return res.json(tokenValidation);
+      }
+      const friendUniqid = await getUserUniqidByNicknameAndTag(db, req.body.friendNickname, req.body.friendsTag)
+      if (!friendUniqid) {
+         return res.json({
+            success: false,
+            msg: 'Could not find user-friend in database'
+         });
+      }
+      if (user.uniqid === friendUniqid) {
+         return res.json({
+            success: false,
+            msg: 'you can not accept yourself as a friend'
+         });
+      }
+      await addFriend(db, user.uniqid, friendUniqid)
+      return res.send({ success: true });
+   })
+
+   app.get('/getFriends', async (req, res) => {
+      if (!req.body.token || !req.body.tokenSelector) {
+         return res.json({
+            success: false,
+            msg: 'incomplete query'
+         });
+      }
+
+      const user = await getUserByTokenSelector(db, req.body.tokenSelector)
+      if (!user) {
+         return res.json({
+            success: false,
+            msg: 'first user not found in database'
+         });
+      }
+      const tokenValidation = await validateUserToken(db, user.uniqid, req.body.token)
```

Comment:
`validateUserToken` fetches the user you already fetched before, we could change the function a little bit so it accepts user's token instead of uniqid (or make a second function that does that) so we don't duplicate database requests

Reply:
Poprawiłem logike i zmergowałem funkcje.